dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -140,4 +124,64 @@ public void getIconData( HttpServletResponse response, @PathVariable String icon
 
         StreamUtils.copyThenCloseInputStream( icon.get().getInputStream(), response.getOutputStream() );
     }
+
+    @GetMapping
+    public @ResponseBody List<IconResponse> getAllIcons( @RequestParam( required = false ) String[] keywords )
+    {
+        Collection<Icon> icons;
+
+        if ( keywords == null )
```

Comment:
Not sure if this is ever happening but can `keywords` be an empty array? If so should we also call `getIcons()` and get all icons?